contracts/token/psp34/psp34.rs

```diff
@@ -206,15 +214,46 @@ impl<T: PSP34Storage + Flush> PSP34Internal for T {
         id: Id,
         data: Vec<u8>,
     ) -> Result<(), PSP34Error> {
-        self._before_token_transfer(&from, &to, &id)?;
+        let owner = self.get().token_owner.get(&id).cloned();
+
+        if owner.is_none() {
```

Comment:
This can be refactored after you add _check_token_exists:
```
        self._before_token_transfer(Some(&from), Some(&to), &id)?;

        let owner = self._check_token_exists(&id)?;

        let caller = Self::env().caller();
        if owner != &caller
            && Some(caller) != self.get().token_approvals.get(&id).cloned()
            && !self._approved_for_all(owner.clone(), caller)
        {
            return Err(PSP34Error::NotApproved)
        }
```